### PR TITLE
CONSOLE-2423: upgrade yaml language server

### LINKED ISSUE #6482: Upgrade yaml-language-server to 0.10.1
currently in 0.9.0 https://github.com/openshift/console/blob/911783a5ab7035b8432013e27ce9fe71e7528d92/frontend/package.json#L166

The 0.10.1 version has been released https://github.com/redhat-developer/yaml-language-server/releases/tag/0.10.1

see changelog for new features https://github.com/redhat-developer/yaml-language-server/blob/0.10.1/CHANGELOG.md#0100

### PR DESCRIPTION
**Pull request by apupier**
not tested locally. Relying on CI at least for a first round.

Closes #6482

### CODE REVIEW

**Comment by spadgett** [comment]
Thanks. We'll need to wait for master to open for 4.7 before we can merge. We're currently in feature freeze, and this doesn't address a specific bug.

/hold

**Comment by apupier** [comment]
```
error: some steps failed:
  * could not run steps: step e2e-gcp-console failed: template pod "e2e-gcp-console" failed: the pod ci-op-d7sysdct/e2e-gcp-console failed after 48m31s (failed containers: test): ContainerFailed one or more containers exited
Container test exited with code 1, reason Error
---
"victory-shared-events@34.3.8" is incompatible with requested version "victory-shared-events@^35.0.8"
warning Resolution field "victory-shared-events@34.3.8" is incompatible with requested version "victory-shared-events@^35.0.8"
warning Resolution field "victory-shared-events@34.3.8" is incompatible with requested version "victory-shared-events@^35.0.8"
warning Resolution field "jquery@3.5.1" is incompatible with requested version "jquery@~3.4.1"
warning Resolution field "minimist@1.2.5" is incompatible with requested version "minimist@0.0.8"
warning Resolution field "minimist@1.2.5" is incompatible with requested version "minimist@0.0.8"
warning Resolution field "minimist@1.2.5" is incompatible with requested version "minimist@~0.0.1"
[3/5] Fetching packages...
warning Pattern ["asn1js@latest"] is trying to unpack in the same destination "/tmp/home/.cache/yarn/v6/npm-asn1js-2.0.26-0a6d435000f556a96c6012969d9704d981b71251-integrity/node_modules/asn1js" as pattern ["asn1js@^2.0.26"]. This could result in non-deterministic behavior, skipping.
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
info There appears to be trouble with your network connection. Retrying...
error An unexpected error occurred: "https://registry.yarnpkg.com/@patternfly/react-tokens/-/react-tokens-4.9.2.tgz: ESOCKETTIMEDOUT".
info If you think this is a bug, please open a bug report with the information provided in "/go/src/github.com/openshift/console/frontend/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

what the exact error?

the incompatible "victory-shared-events" versions? or the connection timeout "An unexpected error occurred: "https://registry.yarnpkg.com/@patternfly/react-tokens/-/react-tokens-4.9.2.tgz: ESOCKETTIMEDOUT"."?

**Comment by spadgett** [comment]
> error An unexpected error occurred: "https://registry.yarnpkg.com/@patternfly/react-tokens/-/react-tokens-4.9.2.tgz: ESOCKETTIMEDOUT".

This was a network error running `yarn install`.

/retest

**Comment by spadgett** [comment]
/assign @yapei @ahardin-rh @reestr 
for QE / docs / PX approval

@ahardin-rh There should be no docs impact here.

**Comment by yapei** [comment]
-  When using `Ctrl + Space` to autocomplete, I can see following js errors
```
Uncaught error 
Error: e.completion.label is undefined
getAriaLabel@https://console-openshift-console.apps.ci-ln-b10qqlb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/static/6-chunk-996ea856c3e53afc4be2.min.js:1:2217938
6218/Z</e.prototype.renderElement@https://console-openshift-console.apps.ci-ln-b10qqlb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/static/6-chunk-996ea856c3e53afc4be2.min.js:1:770708
6218/G</e.prototype.renderElement@https://console-openshift-console.apps.ci-ln-b10qqlb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/static/6-chunk-996ea856c3e53afc4be2.min.js:1:770184
6219/x</e.prototype.insertItemInDOM@https://console-openshift-console.apps.ci-ln-b10qqlb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/static/6-chunk-996ea856c3e53afc4be2.min.js:1:793325
6219/x</e.prototype._splice@https://console-openshift-console.apps.ci-ln-b10qqlb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/static/6-chunk-996ea856c3e53afc4be2.min.js:1:789544
6219/x</e.prototype.splice@https://console-openshift-console.apps.ci-ln-b10qqlb-d5d6b.origin-ci-…
    unexpectedErrorHandler 6-chunk-996ea856c3e53afc4be2.min.js:1
    setTimeout handler*e/this.unexpectedErrorHandler 6-chunk-996ea856c3e53afc4be2.min.js:1
    onUnexpectedError 6-chunk-996ea856c3e53afc4be2.min.js:1
    o 6-chunk-996ea856c3e53afc4be2.min.js:1
    promise callback*8834/Y</e.prototype.onListFocus 6-chunk-996ea856c3e53afc4be2.min.js:1
    e 6-chunk-996ea856c3e53afc4be2.min.js:1
    fire 6-chunk-996ea856c3e53afc4be2.min.js:1
    n 6-chunk-996ea856c3e53afc4be2.min.js:1
    wrapEvent 6-chunk-996ea856c3e53afc4be2.min.js:1
    fire 6-chunk-996ea856c3e53afc4be2.min.js:1
    _set 6-chunk-996ea856c3e53afc4be2.min.js:1
    set 6-chunk-996ea856c3e53afc4be2.min.js:1
    setFocus 6-chunk-996ea856c3e53afc4be2.min.js:1
    showSuggestions 6
.....
```
-  Also when I hover some properties, sometimes the help info are not shown

**Comment by apupier** [comment]
@yapei are these regressions?

linking issues for reference:
https://bugzilla.redhat.com/show_bug.cgi?id=1888870 - JS error when using autocomplete in YAML editor
https://bugzilla.redhat.com/show_bug.cgi?id=1888874 - hover message are not shown for some properties

//cc @evidolob which is the maintainer of Yaml Language Server which will know a lot better than me what is happening.
@evidolob I cannot assign you on the bugzilla issues

**Comment by spadgett** [comment]
@apupier That first bug link was opened specifically for the bug found testing this PR :) Notice the target release is `4.7-premerge`.

**Comment by apupier** [comment]
> That first bug link was opened specifically for the bug found testing this PR

found when testing this PR but was it checked on previous release if it is working or not?

**Comment by yapei** [comment]
@apupier  The version `4.6.0-0.ci.test-2020-10-16-011808-ci-ln-ywvhbm2` is confusing but the cluster was spin up using `clusterbot` so I think the cluster should be in 4.7

**Comment by apupier** [comment]
I would like to know if the issues detected are regressions. it means if the issues are new with this Pull Request or they were already existing in previous versions.
The difference is that if they are new: this is a regression and should block the merge. If they were already existing, it should not block the PR from being merged.

**Comment by evidolob** [comment]
> I would like to know if the issues detected are regressions. it means if the issues are new with this Pull Request or they were already existing in previous versions.

Same for me, as we do not have similar issues on upstream project.

**Comment by ahardin-rh** [comment]
:+1: from the docs side

**Comment by yapei** [comment]
@apupier  Ah I see what you mean, today I launched a cluster with changes in this PR, the two bugs I opened also occurs so these issues already exist, then it is ok for me to merge this PR

**Comment by spadgett** [comment]
> 👍 from the docs side

/label docs-approved

**Comment by spadgett** [comment]
> @apupier Ah I see what you mean, today I launched a cluster with changes in this PR, the two bugs I opened also occurs so these issues already exist, then it is ok for me to merge this PR

Hi, @yapei. I guess I'm confused. Were you testing with the change or without the change? :) I couldn't reproduce the bug against     4.6.0-0.nightly-2020-10-03-051134, but maybe what I tried was a little different.

**Comment by yapei** [comment]
Hey @spadgett  I launched a 4.7.0-0.nightly-2020-10-21-001511 cluster and the issues can be reproduced. Since 4.7.0-0.nightly-2020-10-21-001511 should not have the changes so the faults are not introduced by this PR. That's my assumption. Sorry for the wrong words

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for CEE approval

**Pull request merged**
